pkg/dns/config/config.go

```diff
@@ -117,7 +117,15 @@ func (config *Config) validateUpstreamNameserver() error {
 	}
 
 	for _, ns := range config.UpstreamNameservers {
-		if len(validation.IsValidIP(ns)) > 0 {
+		// TODO(rramkumar): Cleanly split this into seperate function.
+		nsStrings := strings.SplitN(ns, ":", 2)
+		// Validate port if specified
+		if len(nsStrings) == 2 {
+			if _, err := strconv.ParseUint(nsStrings[1], 10, 16); err != nil {
+				return fmt.Errorf("Invalid nameserver: %q", ns)
```

Comment:
nit: first world of error message should be low case.

Reply:
I'm not following... what do should the error message be?

Reply:
fmt.Errorf("invalid nameserver: %q", ns)

glog.Errorf("Invalid ...")

Reply:
Generally speaking, if the purpose of a function is to validate something and return an error, you should not need to log as the error will be logged by the caller up on the call stack. This also avoids spuriously error logging when testing potential configurations.

Reply:
done